internal/dataplane/clients_manager.go

```diff
@@ -189,6 +189,9 @@ func (c *AdminAPIClientsManager) adminAPIAddressNotifyLoop() {
 			// This call will only log errors e.g. during creation of new clients.
 			// If need be we might consider propagating those errors up the stack.
 			c.adjustGatewayClients(discoveredAdminAPIs)
+
+			// Notify subscribers that the clients list has been updated.
+			c.notifyGatewayClientsSubscribers()
```

Comment:
So we are going to notify the client when `AdminAPIClientsManager`  finished to adjust gateway clients, and the subscribers should call `GatewayClients()` on notified? I think we could add one more line of comment to tell how to use the notification.

Reply:
Yeah, exactly. There's already a comment on the public interface `SubscribeToGatewayClientsChanges()` method that explains that: https://github.com/Kong/kubernetes-ingress-controller/blob/d50538bcb7f2b68ad4eef4cd945149e57bff7aaf/internal/dataplane/clients_manager.go#L142.